src/python/pants/backend/nfpm/target_types_rules.py

```diff
@@ -0,0 +1,185 @@
+            },
+            # We use 'dst' as 'src' is not always unique.
+            # This results in an address like: path/to:nfpm_content#/dst/install/path
+            request.template_address.create_generated(dst),
+            union_membership,
+        )
+
+    generated_targets: list[NfpmContentFile] = [generate_tgt(src, dst) for src, dst in src_dst_map]
+
+    if overrides:
+        raise InvalidFieldException(
+            softwrap(
+                f"""
+                Unused key in the `overrides` field for {request.template_address}:
+                {sorted(overrides)}
+                """
+            )
+        )
+
+    return GeneratedTargets(generator, generated_targets)
+
+
+class GenerateTargetsFromNfpmContentSymlinksRequest(GenerateTargetsRequest):
+    generate_from = NfpmContentSymlinks
+
+
+@rule(
+    desc="Generate `nfpm_content_symlink` targets from `nfpm_content_symlinks` target",
+    level=LogLevel.DEBUG,
+)
+async def generate_targets_from_nfpm_content_symlinks(
+    request: GenerateTargetsFromNfpmContentSymlinksRequest,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
+    generator: NfpmContentSymlinks = request.generator
+    # This is not a dict because the same src can be linked to more than one dst.
+    # Also, the field guarantees that there are no dst duplicates in this field.
+    src_dst_map: tuple[tuple[str, str], ...] = generator[NfpmContentSymlinksField].value or ()
+
+    overrides = request.require_unparametrized_overrides()
+
+    def generate_tgt(src: str, dst: str) -> NfpmContentSymlink:
+        tgt_overrides = overrides.pop(dst, {})
+        return NfpmContentSymlink(
+            {
+                **request.template,
+                NfpmContentSymlinkSrcField.alias: src,
+                NfpmContentSymlinkDstField.alias: dst,
+                **tgt_overrides,
+            },
+            # We use 'dst' as 'src' is not always unique.
+            # This results in an address like: path/to:nfpm_content#/dst/install/path
+            request.template_address.create_generated(dst),
+            union_membership,
+        )
+
+    generated_targets: list[NfpmContentSymlink] = [
+        generate_tgt(src, dst) for src, dst in src_dst_map
+    ]
+
+    if overrides:
+        raise InvalidFieldException(
+            softwrap(
+                f"""
+                Unused key in the `overrides` field for {request.template_address}:
+                {sorted(overrides)}
+                """
+            )
+        )
+
+    return GeneratedTargets(generator, generated_targets)
+
+
+class GenerateTargetsFromNfpmContentDirsRequest(GenerateTargetsRequest):
+    generate_from = NfpmContentDirs
+
+
+@rule(
+    desc="Generate `nfpm_content_dir` targets from `nfpm_content_dirs` target",
+    level=LogLevel.DEBUG,
+)
+async def generate_targets_from_nfpm_content_dirs(
+    request: GenerateTargetsFromNfpmContentDirsRequest,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
```

Comment:
Include `nfpm_content_dirs(...)` in at least one of the tests for the `package_nfpm_package` rule.